source/normalize-arguments.js

```diff
@@ -125,6 +132,10 @@ const normalize = (url, options, defaults) => {
 	// Override both null/undefined with default protocol
 	options = merge({path: ''}, url, {protocol: url.protocol || 'https:'}, options);
 
+	for (const hook of options.hooks.init) {
+		hook(options);
```

Comment:
Let's throw a helpful error if the return value of `hook` is a Promise as it means someone accidentally made it async.